### ISSUE TITLE
All Pets: Eligible Pets: 2 ...

### ISSUE THREAD

**Issue by ThistleProse** [tflo/PetWalker#6]
Hello!

I'm not sure why, but PW is only pulling 2 pets, from my pet pool. I have 1,014 pets, and 814 are unique... but it's only summoning one of two pets, and I don't know why?

The two pets are 'Firebat Pup' and 'Corgnelius', neither of which are favourites.

What am I doing wrong??

> -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
PetWalker: Status & Settings:
[16:35:03] 
Automatic Random-summoning / Restore is enabled.
Summon Timer is 12 minutes. Next random pet in 11:34.
Automatic summoning while mounted in a Dragonriding zone is not allowed.
Verbosity level of messages: 3 (of 3).
Pet Pool is set to All Pets. Eligible pets: 2.
Per-character Favorites are enabled for Nimueh.
[16:35:03] 
Nimueh has 0 character-specific favorite pets. 
-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --

**Comment by tflo** [comment]
Thanks for posting the Status output.

The "Eligible pets: 2" means that the pet pool is limited to 2 pets, which is what you are experiencing. The "Pet Pool is set to All Pets" means that you are not using favorites.

So, in theory, the only cause could be the filter settings in the Pet Journal. These filters can actually restrict the pet pool, i.e. you can actively use them to "shape" your pet pool without the need for favorites. This includes any text in the search box.

However, it is also possible for the game to mess up these settings (especially after a hard disconnect), or you may have been searching for a pet and forgotten to reset the filters.

For example, I can reproduce your Corgnelius and Firebat Pup pet pool with these filter/search settings:

<img width="787" alt="WoWScrnShot_040623_141353" src="https://user-images.githubusercontent.com/1410282/230379798-fb79d935-aa09-4c97-a74c-3a532256493e.png">

In other words, if you want to have truly _all_ pets as random pool, make sure that there is no "X" badge <img width="25" alt="WoWScrnShot_040623_141353 copy" src="https://user-images.githubusercontent.com/1410282/230380407-61c1845b-7583-4ec9-a178-f37e8197d709.png"> on the filter settings[^1] and that the search box is empty.

---

If this is not the reason for the issue, please let me know which version of PW you are using (the latest one is [1.2.2](https://www.curseforge.com/wow/addons/petwalker/files/4468110) from 31 March).

[^1]: Exception: The "Not Collected" filter is irrelevant for the pool.

**Comment by ThistleProse** [comment]
Hello!

thank you -- I didn't have any search filtering set on Rematch, _but_ when I turned off Rematch (see pic), I discovered that I did, indeed, have the search filter set up. I feel a bit silly, and thank you for your time! My pet pool now consists of 1000 pets :D

> -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --
PetWalker: Status & Settings:
[14:51:26] 
Automatic Random-summoning / Restore is enabled.
Summon Timer is 12 minutes. Next random pet in 0:00.
Automatic summoning while mounted in a Dragonriding zone is not allowed.
Verbosity level of messages: 3 (of 3).
Pet Pool is set to All Pets. Eligible pets: 1005.
Per-character Favorites are disabled for Varlea.
[14:51:26] 
Varlea has 0 character-specific favorite pets. 
-- -- -- -- -- -- -- -- -- -- -- -- -- -- -- -- --

![image](https://user-images.githubusercontent.com/28851048/230543138-5939a3a5-d51e-4ffb-a1b3-fd3717fb0b97.png)

**Comment by tflo** [comment]
> I feel a bit silly

No need to. Thanks for bringing the issue to my attention. I'm using separate shortcuts for Rematch and Blizz's pet journal (and with Use Default Pet Journal enabled), so I wasn't really aware of that potential pitfall.

I've added a FAQ point to the readme and descriptions on CF/Wago to be more explicit on that, and also modified the in-game Pet Pool warning message.

– Tom
